### PR TITLE
[stateless_validation] Refactor ChunkEndorsementsState

### PR DESCRIPTION
**Pull request by shreyan-gupta**
I noticed in endorsement tracker, we were first checking if we have enough endorsement stake and then using that information to generate the signature vector. 

It's simpler and more efficient to generate this while calculating enough stake.

ChunkEndorsementsState doesn't have to be an enum, we can include the signatures in EndorsementStats (and rename it to ChunkEndorsementsState).

### CODE REVIEW

File: chain/client/src/stateless_validation/chunk_endorsement/tracker_v2.rs [hunk 1]
```diff
@@ -77,8 +72,10 @@ impl ChunkEndorsementTracker {
             self.epoch_manager.get_epoch_id_from_prev_block(chunk_header.prev_block_hash())?;
         let protocol_version = self.epoch_manager.get_epoch_protocol_version(&epoch_id)?;
         if !ProtocolFeature::StatelessValidation.enabled(protocol_version) {
-            // Return an empty array of chunk endorsements for older protocol versions.
-            return Ok(ChunkEndorsementsState::Endorsed(None, vec![]));
+            // Return an endorsed empty array of chunk endorsements for older protocol versions.
+            let mut endorsement_stats = ChunkEndorsementsState::default();
```
Comment (pugachAG) [inline chain/client/src/stateless_validation/chunk_endorsement/tracker_v2.rs:76]: rust nit: 
```
ChunkEndorsementsState {
  is_endorsed: true,
  ..ChunkEndorsementsState::default()
}
```

File: core/primitives/src/stateless_validation/validator_assignment.rs [hunk 2]
```diff
@@ -51,23 +55,35 @@ impl ChunkValidatorAssignments {
...
+                    signatures.push(Some(Box::new(signature)));
+                }
+                None => signatures.push(None),
             }
         }
-        EndorsementStats {
+        // Signatures are empty if the chunk is not endorsed
+        let is_endorsed = has_enough_stake(total_stake, endorsed_stake);
+        if !is_endorsed {
+            signatures = vec![];
```
Comment (pugachAG) [inline core/primitives/src/stateless_validation/validator_assignment.rs:78]: nit: `signatures.clear()`

File: core/primitives/src/stateless_validation/validator_assignment.rs [hunk 3]
```diff
@@ -51,23 +55,35 @@ impl ChunkValidatorAssignments {
 
     pub fn compute_endorsement_stats(
```
Comment (pugachAG) [inline core/primitives/src/stateless_validation/validator_assignment.rs:56]: nit: also update method name

File: chain/client/src/stateless_validation/chunk_endorsement/tracker_v1.rs [hunk 4]
```diff
@@ -209,7 +203,9 @@ impl ChunkEndorsementTrackerInner {
         let protocol_version = self.epoch_manager.get_epoch_protocol_version(&epoch_id)?;
         if !checked_feature!("stable", StatelessValidation, protocol_version) {
             // Return an empty array of chunk endorsements for older protocol versions.
-            return Ok(ChunkEndorsementsState::Endorsed(None, vec![]));
+            let mut endorsement_stats = ChunkEndorsementsState::default();
```
Comment (pugachAG) [inline chain/client/src/stateless_validation/chunk_endorsement/tracker_v1.rs:206]: this is used multiple times, please extract into a separate method, something like `ChunkEndorsementsState::stateless_validation_disabled()` or `ChunkEndorsementsState::old_protocol_version()`

**Review by pugachAG — APPROVED** [review]

File: chain/client/src/stateless_validation/chunk_endorsement/tracker_v1.rs (hunk 4 above)
Comment (shreyan-gupta) [reply chain/client/src/stateless_validation/chunk_endorsement/tracker_v1.rs:206]: With a future PR, we'll get rid of tracker_v1 and we'll have only one instance of this pattern left.

**Pull request merged**